File: src/main/java/sage/transfer/NotifView.java
package sage.transfer;

import java.util.Date;

import sage.entity.Notification;

public class NotifView {
  public long id;
  public long ownerId;
  public UserLabel sender;
  public Date time;
  public String type;
  public String desc;
  public String source;
  public boolean read = false;

  NotifView() {}

  public NotifView(Notification notification, UserLabel sender, String source) {
    id = notification.getId();
    ownerId = notification.getOwnerId();
    this.sender = sender;
    time = notification.getWhenCreated();
    type = notification.getType().name();
    desc = notification.getType().getDesc();
    this.source = source;
  }
}

File: src/main/kotlin/sage/transfer/NotifView.kt
package sage.transfer

import sage.annotation.KotlinNoArg
import java.util.Date

import sage.entity.Notification

@KotlinNoArg
class NotifView {
  var id: Long = 0
  var ownerId: Long = 0
  var sender: UserLabel? = null
  var time: Date? = null
  var type: String  = ""
  var desc: String = ""
  var source: String = ""
  var read = false

  constructor(notification: Notification, sender: UserLabel, source: String) {
    id = notification.id
    ownerId = notification.ownerId!!
    this.sender = sender
    time = notification.whenCreated
    type = notification.type!!.name
    desc = notification.type!!.desc
    this.source = source
  }
}
